### PR TITLE
Test: queued operation is not serialized

### PR DESCRIPTION
**Pull request by vweevers**
@ralphtheninja anything else we should test?

### CODE REVIEW

**Comment by vweevers** [comment]
Perhaps a stream + json encoding?

File: test/json-test.js [hunk 1]
```diff
@@ -7,18 +7,20 @@ var levelup = require('../lib/levelup.js')
...
 
-buster.testCase('JSON API', {
+buster.testCase('JSON encoding', {
   'setUp': function (done) {
     common.commonSetUp.call(this, function () {
       this.runTest = function (testData, assertType, done) {
         var location = common.nextLocation()
         this.cleanupDirs.push(location)
         levelup(encDown(leveldown(location), {
+          keyEncoding: 'json',
```
Comment (vweevers) [inline test/json-test.js]: Thanks to the added stream test, I found out that these tests were actually using serialized keys (`'[object Object]'`).

(this is not related to `deferred-leveldown`)

File: test/deferred-open-test.js [hunk 2]
```diff
@@ -169,5 +169,24 @@ buster.testCase('Deferred open()', {
         }
       })
     }
+  },
+
+  'queued operation is not serialized': function (done) {
+    var location = common.nextLocation()
+    var db = levelup(encDown(leveldown(location), { valueEncoding: 'json' }))
```
Comment (ralphtheninja) [inline test/deferred-open-test.js]: You could do `{ keyEncoding: 'json' }` and also test the key

Comment (vweevers) [reply test/deferred-open-test.js]: Added a test.

File: test/deferred-open-test.js [hunk 3]
```diff
@@ -188,5 +189,23 @@ buster.testCase('Deferred open()', {
...
+
+    this.closeableDatabases.push(db)
+    this.cleanupDirs.push(location)
+
+    // deferred-leveldown < 2.0.2 would serialize the key to a string.
+    db.put({ thing: 2 }, 'value', function (err) {
+      refute(err)
+
+      db.createKeyStream().pipe(concat(function (result) {
+        assert.equals(result, [{ thing: 2 }])
```
Comment (ralphtheninja) [inline test/deferred-open-test.js]: :+1:
